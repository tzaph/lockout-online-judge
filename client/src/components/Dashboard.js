import React, { useEffect, useState } from "react";
import { useAuth } from "../contexts/AuthContext";
import { Link, useNavigate } from "react-router-dom";
import { Alert } from "react-bootstrap";
import { getDatabase, ref, get, child } from "firebase/database";

export default function Dashboard() {
  const [error, setError] = useState("");
  const [data, setData] = useState({});
  const currentUser = useAuth()?.currentUser;
  const logout = useAuth()?.logout;
  const navigate = useNavigate();

  const handleLogout = async () => {
    setError("");
    try {
      await logout();
      setError("");
      navigate("/login");
    } catch {
      setError("Failed to logout");
    }
  };

  const Getdata = () => {
    get(child(ref(getDatabase()), "users/" + currentUser.uid))
      .then((snapshot) => {
        if (snapshot.exists()) {
          setData(snapshot.val());
        } else {
        }
      })
      .catch((error) => {
        console.error(error);
      });
  };
  useEffect(() => {
    Getdata();
  }, []);

  return (
    <div className="dashboard-container">
      <h1>Dashboard</h1>
      <div className="profile-container">
        <h2>Profile</h2>
        <label>Email: {data.email}</label>
        <label>Name: {data.name}</label>
        <label>Codeforces Handle: {data.codeforcesHandle}</label>
        <label>Rating: {data.rating}</label>
        <br></br>
        <Link to="/update-profile">Update Profile</Link>
        {data.admin ? (
          <Link to="/refresh-problem-database">Refresh Problem Database</Link>
        ) : null}
      </div>
      {error && <Alert variant="danger">{error}</Alert>}
      <button className="button" onClick={handleLogout}>
        Logout
      </button>
    </div>
  );
}
