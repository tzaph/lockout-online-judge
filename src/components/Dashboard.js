import React, { useState } from 'react'
import { useAuth } from '../contexts/AuthContext';
import { Link, useNavigate } from 'react-router-dom';

export default function Dashboard() {
  const [error, setError] = useState('');
  const { currentUser, logout } = useAuth();
  const navigate = useNavigate();

  const handleLogout = async() => {
    setError('');
    try {
      await logout();
      setError('');
      navigate('/login');
    } catch {
      setError('Failed to logout');
    }    
  }

  return (
    <>
      <div>Dashboard</div>
      <div className="profile-container">
        <h2>Profile</h2>
        Email: {currentUser.email}
        Name: {currentUser.displayName}
        <Link to="/update-profile">Update Profile</Link>
      </div>
      <button className="button" onClick={handleLogout}>Logout</button>
    </>
  );
}