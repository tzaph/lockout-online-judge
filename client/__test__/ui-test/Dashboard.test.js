import React from "react";
import renderer from "react-test-renderer";
import Dashboard from "../../src/components/Dashboard";
import { BrowserRouter as Router } from 'react-router-dom';

describe("<Dashboard />", () => {
  it("Renders correctly", () => {
    const tree = renderer.create(<Router><Dashboard /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
