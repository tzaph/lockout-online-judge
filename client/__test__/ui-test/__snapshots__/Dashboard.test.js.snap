// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> Renders correctly 1`] = `
<div
  className="dashboard-container"
>
  <h1>
    Dashboard
  </h1>
  <div
    className="profile-container"
  >
    <h2>
      Profile
    </h2>
    <table>
      <tr
        align="left"
      >
        <td>
          Email:
        </td>
        <td />
      </tr>
      <tr
        align="left"
      >
        <td>
          Name:
        </td>
        <td />
      </tr>
      <tr
        align="left"
      >
        <td>
          Codeforces Handle:
        </td>
        <td />
      </tr>
      <tr
        align="left"
      >
        <td>
          Rating:
        </td>
        <td />
      </tr>
    </table>
    <br />
    <a
      href="/update-profile"
      onClick={[Function]}
    >
      Update Profile
    </a>
  </div>
  <button
    className="button"
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
